core/templates/dev/head/pages/topic-editor-page/modal-templates/select-skill-modal.template.html

```diff
@@ -1,31 +1,76 @@
+    <h3 ng-if="skillSummaries.length > 0">
+      Select the skill to link the question to:
+    </h3>
+  </div>
+  <div class="modal-body select-skill-modal">
+    <div ng-if="skillSummaries.length == 0">
+      You can only create questions for topics that have atleast one skill
+      attached to it. Please link a skill to this topic before adding questions.
+    </div>
+    <div class="list-group">
+      <a class="list-group-item protractor-test-skill-item"
+         ng-repeat="skillSummary in skillSummaries track by $index"
+         ng-click="selectOrDeselectSkill(skillSummary.getId(), $index)"
+         ng-class="{'list-group-item-success': skillSummary.isSelected}">
+        <label class="list-group-item-heading" style="font-size: 1em;"><[skillSummary.getDescription()]></label>
+      </a>
+    </div>
   </div>
-  <div class="list-group">
-    <a class="list-group-item protractor-test-skill-item"
-       ng-repeat="skillSummary in skillSummaries track by $index"
-       ng-click="selectOrDeselectSkill(skillSummary.getId(), $index)"
-       ng-class="{'list-group-item-success': skillSummary.isSelected}">
-      <label class="list-group-item-heading" style="font-size: 1em;"><[skillSummary.getDescription()]></label>
-    </a>
+  <div class="modal-footer" ng-switch="skillSummaries.length > 0">
+    <div ng-switch-when="true">
+      <button class="btn btn-default" ng-click="cancelModal()"> Cancel </button>
+      <button class="btn btn-success protractor-test-confirm-skill-button"
+              ng-disabled="selectedSkillIds.length === 0"
+              ng-click="goToSelectDifficultyView()"> Next
+      </button>
+    </div>
+    <div ng-switch-when="false">
+      <button class="btn btn-default" ng-click="closeModal()"> Ok </button>
+    </div>
   </div>
 </div>
-<div class="modal-footer" ng-switch="skillSummaries.length > 0">
-  <div ng-switch-when="true">
-    <button class="btn btn-default" ng-click="cancel()"> Cancel </button>
-    <button class="btn btn-success protractor-test-confirm-skill-button" ng-disabled="selectedSkillId === null" ng-click="done()"> Done </button>
+<!-- Select Difficulty View -->
+<div ng-if="currentMode === 'MODE_SELECT_DIFFICULTY'">
+  <div class="modal-header">
+    <h3 ng-if="linkedSkillsWithDifficulty.length == 0">
+      Select skills you would like to link to the question.
+    </h3>
+    <h3 ng-if="linkedSkillsWithDifficulty.length > 0">
+      Select the difficulty per skill:
+    </h3>
+  </div>
+  <div class="modal-body select-skill-modal">
+    <div ng-if="linkedSkillsWithDifficulty.length == 0">
+      You can only set difficulty after selecting skills linked to the question. Please go back and link skill(s) to this question.
+    </div>
+    <div class="list-group">
+      <div class="list-group-item"  ng-repeat="skillWithDifficulty in linkedSkillsWithDifficulty track by $index">
+        <label class="list-group-item-heading" style="font-size: 1em;"><[skillWithDifficulty._description]></label>
```

Comment:
Here and elsewhere: getDescription(), getDifficulty(), etc. rather than referring to private properties.

Reply:
I could do getDescription, but getDifficulty() won't work here because I am modifying difficulty based on the radio button click so I do need to pass the object variable